src/sentry/dynamic_sampling/tasks.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Mapping, Sequence
+
+from sentry import features
+from sentry.dynamic_sampling.models.adjustment_models import AdjustedModel
+from sentry.dynamic_sampling.models.adjustment_models import Project as DSProject
+from sentry.dynamic_sampling.prioritise_projects import fetch_projects_with_total_volumes
+from sentry.dynamic_sampling.rules.helpers.prioritise_project import _generate_cache_key
+from sentry.dynamic_sampling.rules.helpers.utils import get_redis_client_for_ds
+from sentry.models import Organization
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.relay import schedule_invalidate_project_config
+from sentry.utils import metrics
+
+CHUNK_SIZE = 1000
+MAX_SECONDS = 60
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.prioritise_projects",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def prioritise_projects() -> None:
+    metrics.incr("sentry.tasks.dynamic_sampling.prioritise_projects.start", sample_rate=1.0)
+    with metrics.timer("sentry.tasks.dynamic_sampling.prioritise_projects", sample_rate=1.0):
+        for org_id, project_id_with_count_per_root in fetch_projects_with_total_volumes().items():
+            process_projects_sample_rates.delay(org_id, project_id_with_count_per_root)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.process_projects_sample_rates",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def process_projects_sample_rates(
+    organization_id: int, project_id_with_count_per_root: Sequence[Mapping[int, int]]
+) -> None:
+    """
+    Takes a single org id and a list of project ids
+    """
+    organization = Organization.objects.get_from_cache(id=organization_id)
+    # Check if feature flag is enabled:
+    if features.has("organizations:ds-prioritise-by-project-bias", organization):
+        with metrics.timer("sentry.tasks.dynamic_sampling.process_projects_sample_rates.core"):
+            adjust_sample_rates(organization_id, project_id_with_count_per_root)
+
+
+def adjust_sample_rates(
+    org_id: int, project_id_with_count_per_root: Sequence[Mapping[int, int]]
+) -> None:
+    """
+    This function apply model and adjust sample rate per project in org
+    and store it in DS redis cluster, then we invalidate project config
+    so relay can reread it, and we'll inject it from redis cache.
+    """
+    projects = []
+    for project_id, count_per_root in project_id_with_count_per_root:
+        projects.append(
+            DSProject(id=project_id, count_per_root=count_per_root, blended_sample_rate=0.0)
```

Comment:
Is there any reason for `blended_sample_rate=0.0`?

Reply:
fixed.